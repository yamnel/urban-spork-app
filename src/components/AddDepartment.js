import React from 'react';
import {Col, Form, FormGroup, Input, Label} from 'reactstrap';



export default class AddDepartment extends React.Component {

    constructor(){
        super();

        this.state = {
            InputPlaceholder: 'Enter department name',
        };

        // this.onInputChange = this.onInputChange.bind(this);

    }


    onInputChange = (data) => {
        this.props.AddButton(data.target.value);

    };

    render() {
        return(
            <div>
                <Form>
                    <FormGroup row>
                        <Col md={6}>
                            <Label color={"muted"} for={"Name"}>
                                Department Name:
                            </Label>
                        </Col>

                        <Col md={6} >
                            <Input placeholder={this.state.InputPlaceholder} id={"Name"} onChange={evt => {this.onInputChange(evt)}}/>
                        </Col>
                    </FormGroup>

                </Form>
            </div>
        )
    }

}



