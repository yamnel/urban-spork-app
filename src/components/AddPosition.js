import React from 'react';
import {Col, Form, FormGroup, Input, Label} from 'reactstrap';



export default class AddPosition extends React.Component {

    constructor(){
        super();

        this.state = {
            InputPlaceholder: 'Enter title of position',
            department:"",
        };

        this.onInputChange = this.onInputChange.bind(this);
        this.updateDepartment = this.updateDepartment.bind(this);

    }


    onInputChange = (data) => {
        console.log("onInputChange");
        this.props.AddButton(data.target.value);

    };

    updateDepartment = (department) => {

        this.props.DepartmentSelected(department);

    };

    render() {
        return(
            <div>
                <Form>
                    <FormGroup row>
                        <Col md={4}>
                            <Label color={"muted"}  for={"Title"}>
                                Position Title:
                            </Label>
                        </Col>

                        <Col md={8}>
                            <Input placeholder={this.state.InputPlaceholder} id={"Title"} onChange={evt => {this.onInputChange(evt)}}/>
                        </Col>
                    </FormGroup>
                    <br/>
                    <FormGroup row>
                        <Col md={6}>
                            <Label for="SelectDepartment">
                                Select Department
                            </Label>
                        </Col>

                        <Col md={6}>
                            <Input type="select"  id="SelectDepartment" onChange={e => {this.updateDepartment(e)}}>
                                <option></option>
                                <option>Warehouse</option>
                                <option>Marketing</option>
                                <option>Human Resource</option>
                                <option>Complaints</option>
                            </Input>
                        </Col>

                    </FormGroup>

                </Form>
            </div>
        )
    }

}



