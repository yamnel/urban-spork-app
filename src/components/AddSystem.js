import React from 'react';
import {Col, Form, FormFeedback, FormGroup, Input, Label} from 'reactstrap';


export default class AddDepartment extends React.Component {

    constructor() {
        super();

        this.state = {
            DescriptionPlaceholder: 'Enter System Description',
            LogoURLPlaceholder: 'Logo Image URL HERE',
        };
    }

    render() {
        return (
            <div>
                <Form>
                    <FormGroup row>
                        <Col md={6}>
                            <Label color={"muted"} for={"SystemName"}>
                                System Name:
                            </Label>
                        </Col>

                        <Col md={6}>
                            <Input placeholder={"Enter System Name"} id={"systemName"} onChange={evt => {
                                this.props.onInputChange(evt)

                            }}/>
                            <FormFeedback>This feeld has to be filled</FormFeedback>
                        </Col>
                    </FormGroup>

                    <FormGroup row>
                        <Col md={6}>
                            <Label color={"muted"} for={"SystemDescription"}>
                                System Description:
                            </Label>
                        </Col>

                        <Col md={6}>
                            <Input placeholder={this.state.DescriptionPlaceholder} id={"systemDescription"}
                                   onChange={evt => {
                                       this.props.onInputChange(evt)
                                   }}/>
                        </Col>
                    </FormGroup>

                    <FormGroup row>
                        <Col md={6}>
                            <Label color={"muted"} for={"SystemLogoURL"}>
                                System Logo URL:
                            </Label>
                        </Col>

                        <Col md={6}>
                            <Input placeholder={this.state.LogoURLPlaceholder} id={"systemLogoURL"} onChange={evt => {
                                this.props.onInputChange(evt)
                            }}/>
                        </Col>
                    </FormGroup>

                </Form>
            </div>
        )
    }

}



