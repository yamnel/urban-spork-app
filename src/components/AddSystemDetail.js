import React from 'react';
import {Col, Form, FormGroup, Input, Label, Button} from 'reactstrap';
import Dropzone from 'react-dropzone'


export default class AddSystemDetailComponent extends React.Component {
    //state = {...this.props.userData};
    state = {
        dropZoneDisabled: true,
        files: []
    };

   // editedData = {...this.state};

    // handleOnChange = (e) => {
    //     console.log(e.target)
    // };

    handleAddSystem = () => {
    };

    render() {
        return(
            <div>
                <Form>
                    <FormGroup row>
                        <Label color={"muted"} sm={"3"} for={"systemName"}>
                            System Name:
                        </Label>
                        <Col sm={20}>
                            <Input id={"systemName"} onChange={this.onInputChange}/>
                        </Col>
                    </FormGroup>

                    <FormGroup row>
                        <Label color={"muted"} sm={"3"} for={"systemDescription"}>
                             System Description:
                        </Label>
                        <Col sm={20}>
                            <Input id={"systemDescription"}/>
                        </Col>
                    </FormGroup>

                    <FormGroup row>
                        <Label color={"muted"} sm={"3"} for={"systemDescription"}>
                            System Logo URL:
                        </Label>
                        <Col sm={20}>
                            <Input id={"systemLogo"}/>
                        </Col>
                    </FormGroup>
                </Form>
            </div>
        )
    }

}



