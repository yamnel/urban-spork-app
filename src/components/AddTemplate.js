import React from 'react';
import {Col, Form, FormGroup, Input, Label} from 'reactstrap';
import PermissionMultiselectComponent from "./PermissionMultiselectComponent";

export default class AddTemplate extends React.Component {

    state = {
        Permissions: [],
        TemplateToAdd: {
            Name: "",
            TemplatePermissions: {

            }
        }
    }

    onInputChange = (e) => {
        this.props.templateTitle(e.target.value);
    };

    onPermissionsChanged = (e) => {
        this.setState({Permissions: e});
        this.props.templatePermissions(e);
    }


    render(){
        return(
            <div>
                <Form>
                    <FormGroup row>
                        <Col md={6}>
                            <Label for="TemplateName">
                                Template Name
                            </Label>
                        </Col>

                        <Col md={6}>
                            <Input placeholder={this.state.InputPlaceholder} id={"TemplateName"} onChange={e => {this.onInputChange(e)}}/>
                        </Col>

                        <Col md={6}>
                            <br/>
                            <Label for="PermissionSelect">
                                Permissions
                            </Label>
                        </Col>

                        <Col md={5}>
                            <br/>
                            <PermissionMultiselectComponent id={"PermissionSelect"} selectedPermissions = {this.state.Permissions}
                                setPermissions = {e => {this.onPermissionsChanged(e)}}/>
                        </Col>

                    </FormGroup>

                </Form>
            </div>
        );
    }
}