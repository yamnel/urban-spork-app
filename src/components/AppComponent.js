import React from 'react';
import {connect} from "react-redux";

import AppRouter from "../routers/AppRouter";
import LogInPage from "./LogInPage";
import {getUsersData} from "../actions/users";
import {setManagerName, setManagerId} from "../actions/manager";
import {getAllPermissions} from "../actions/permissions";
import NonAdminRouter from "../routers/NonAdminRouter";


class AppComponent extends React.Component{

    state = {
        logIn: true,
        isAdmin: undefined,
        isNonAdmin: undefined,
        allowAccess: true,
        managerId: undefined
    };


    accessGranted = () => {
        // this.setState(() => ({allowAccess: true, managerId}));
        this.props.getUserData();
        this.props.getAllPermissions();
    };


    isAdmin = (b, id, name) => {
        console.log(b);

        this.props.setManagerId(id);
        this.props.setManagerName(name);

        if (b == "true") {

            this.setState({isAdmin: true});
            this.setState({isNonAdmin: false});
            this.setState({logIn: false});
        } else {
            console.log("Non-Admin User");

            this.setState({isNonAdmin: true});
            this.setState({isAdmin: false});
            this.setState({logIn: false});
        }
    };



    render() {
        return (
            <div>
                {}
                {this.state.logIn  && <LogInPage isAdmin={this.isAdmin}/> }
                {this.state.isAdmin && ( <AppRouter grantAccess={this.accessGranted()} /> )}
                {this.state.isNonAdmin && (<NonAdminRouter/>)}
                {/*{this.accessGranted()}*/}
                {/*{this.state.allowAccess? ( <AppRouter /> ) : ( <LogInPage accessGranted={this.accessGranted}/>)}*/}
            </div>
        )
    }

}

const mapDispatchToProps = (dispatch)  => ({
    getUserData: () => dispatch(getUsersData()),
    setManagerId: (id) => dispatch(setManagerId(id)),
    setManagerName: (name) => dispatch(setManagerName(name)),
    getAllPermissions: () => dispatch(getAllPermissions()),
});

export default connect(undefined,mapDispatchToProps)(AppComponent)