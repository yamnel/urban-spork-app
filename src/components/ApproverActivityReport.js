import React from 'react';
import UserTable from "./UserTable";
import UrbanSporkAPI from '../api/UrbanSporkAPI';
import moment from 'moment';
import {Row,Col,Input,Button,Table} from "reactstrap";
import jsPDF from 'jspdf'
import ReactDOMServer from 'react-dom/server';
import FontAwesomeIcon from '@fortawesome/react-fontawesome';
import {faFilePdf} from '@fortawesome/fontawesome-free-solid';




export default class ApproverActivityReport extends React.Component {
    state = {
        Data: [],
        AdminList: []
    };

    componentWillMount() {
        this.getData();
        this.getAdminDropDown();
    }

    columns = [
        {accessor: 'permissionName', Header: 'System'},
        {accessor: 'truncatedEventType', Header: 'Activity'},
        {accessor: 'forFullName', Header: 'For'},
        {accessor: 'approverFullName', Header: 'By'},
        {
            accessor: 'timeStamp',
            Header: 'Date',
            Cell: ({value}) => moment.utc(value).format('ddd MMM D YYYY').toString()
        },
    ];

    styles = {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
        textAlign: 'center'
    };

    getData = () => {
        const Data = UrbanSporkAPI.getApproverActivity();
        Data.then(data => this.setState({Data: data})).catch(() => this.setState({Data: []}));
    };


    getAdminDropDown = () => {
        const Admins = UrbanSporkAPI.getApproverDropDown();
        Admins.then(data => this.setState({AdminList: data})).catch(() => this.setState({AdminList: []}));
    };

    updateTable = (admin) => {
        console.log(admin.target.value);
        let payload = {
            ApproverId: admin.target.value,
        };
        const newData = UrbanSporkAPI.getApproverActivityReport(payload);
        newData.then(data => this.setState({Data: data})).catch(() => this.setState({Data: []}));

    };



    formatReport = () => {
        let pdfBody = this.state.Data.map((entry, index) => (
            <tr key={index}>
                <td>{index + 1}</td>
                <td>{entry.permissionName}</td>
                <td>{entry.truncatedEventType}</td>
                <td>{entry.forFullName}</td>
                <td>{entry.byFullName}</td>
                <td>{entry.timeStamp}</td>
            </tr>
        ));
        return pdfBody;
    };

    exportToPDF = () => {
        let pdf = new jsPDF('p', 'pt', 'letter');
        let rows = this.formatReport();
        let table = (
            <Table >
                <thead style={{height:20}}>

                <th>#</th>
                <th>System</th>
                <th>Activity</th>
                <th>For</th>
                <th>By</th>
                <th>Date</th>

                </thead>
                <tbody>
                {rows}
                </tbody>
            </Table>
        );

        let page = (
            <Row>
                <Row>
                    <Col md={4}>

                    </Col>
                    <Col md={4} textalign="center">
                        <h1>Approver Activity Report</h1>
                    </Col>
                    <Col md={4}>

                    </Col>
                </Row>
                <Row>
                    {table}
                </Row>
            </Row>
        );
        let source = ReactDOMServer.renderToStaticMarkup(page);

        let margins = {
            top: 50,
            left: 60,
            width: 545
        };

        pdf.fromHTML(
            source // HTML string or DOM elem ref.
            , margins.left // x coord
            , margins.top // y coord
            , {
                'width': margins.width // max width of content on PDF
                ,
            },
            function (dispose) {
                // dispose: object with X, Y of the last line add to the PDF
                // this allow the insertion of new lines after html
                pdf.save('ActivityReport.pdf');
            }
        );
    };

    render() {
        const AllAdmins = this.state.AdminList.map((Admin, index) => (

            <option value={Admin.id} key={index} >{Admin.name}</option>

        ));
        return (
            <div>
                <div >
                    <Row style={{height: 50}}>
                    </Row>
                    <Row style={this.styles}>
                        <Col md={3}>
                            <Input type="select"  id="SelectApprover" onChange={e => {this.updateTable(e)}}>
                                <option>
                                    Select Admin/Approver
                                </option>
                                {AllAdmins}
                            </Input>
                        </Col>
                        <Col md={6}>
                            <h1>Approver Activity Report</h1>
                        </Col>
                        <Col md={3}>
                            <Button onClick={() => {this.exportToPDF()}}>
                                <FontAwesomeIcon icon={faFilePdf}/>


                            </Button>
                        </Col>
                    </Row>
                </div>
                <div>
                    <UserTable columns={this.columns} data={this.state.Data}/>
                </div>
            </div>
        )
    }
}