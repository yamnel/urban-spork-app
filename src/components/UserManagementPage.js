import React from 'react';
import UserTable from "./UserTable";
import {Link} from 'react-router-dom';
import {Button, FormGroup, Input} from 'reactstrap';
import {connect} from "react-redux";
import selectUsers from "../selectors/users";
import {setTextFilter} from "../actions/filters";
import UserDetailsModal from "./UserDetailsModal";

export class UserManagementPage extends React.Component {
    // Last ID used stays in the state, I don't like it but it's a bug I need to fix later.
    state = {
        modalIsOpen: false
    };


    columns = [
        // {accessor: 'id', Header: 'User ID'},
        {accessor: 'name', Header: 'Name'},
        {accessor: 'email', Header: 'Email'},
        {accessor: 'department', Header: 'Department'},
        {accessor: 'title', Header: 'Title'}

    ];



    // componentDidMount() {
    //     getUsersData();
    //
    // }

    openUserDetailsModal = (selectedUserData) => {
        this.setUserData(selectedUserData.id);
        this.toggleModalIsOpen();

        //proof that is stays in the old state
        // console.log(`The new state field selectedUserId is ${this.state.selectedUserId}`);
    };

    // onCreateUserClick = () =>{
    //
    // };


    toggleModalIsOpen = () => this.setState(() => ({modalIsOpen: !this.state.modalIsOpen}));


    // Where do I call this?
    // unsetUserData = () => {
    //     console.log('Should be Undef.');
    //     this.setState(() => ({selectedUserId: undefined}))
    // };

    setUserData = (selectedUserId) => this.setState(() => ({selectedUser: selectedUserId}));

    onInputChange = (e) => {
        this.props.setTextFilter(e.target.value);
    };

    render() {
        return (
            <div>
                <div className={'user-management-nav'}>
                    <Button to={"/create-user"} color={'success'} tag={Link}>Create User</Button>
                    {/*<Button color={'primary'}>Filter</Button>*/}

                    <FormGroup>
                        <Input type={'text'} value={this.props.filters.text} onChange={this.onInputChange} placeholder={'Search'} />
                    </FormGroup>
                </div>

                <div>
                    <UserTable onRowClick={this.openUserDetailsModal} data={this.props.users} columns={this.columns}/>
                </div>

                <UserDetailsModal isOpen={this.state.modalIsOpen} toggle={this.toggleModalIsOpen} userId={this.state.selectedUser}/>
            </div>
        )
    }

}

const mapStateToProps = (state) => {
    return {
        users: selectUsers(state.users, state.filters),
        filters: state.filters
    }
};

const mapDispatchToProps = (dispatch)  => ({
    setTextFilter: (text) => dispatch(setTextFilter(text)),
    // getUserData: () => dispatch(getUsersData())
});


export default connect(mapStateToProps, mapDispatchToProps)(UserManagementPage);

