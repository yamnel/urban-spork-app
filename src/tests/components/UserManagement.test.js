import React from 'react';
import {shallow} from 'enzyme';
import {filters, altFilters} from '../fixtures/filters';
import {UserManagementPage} from "../../components/UserManagementPage";

let wrapper, setTextFilter, history;

beforeEach(() => {
    setTextFilter = jest.fn();
    history = {push: jest.fn()};

    wrapper = shallow(<UserManagementPage
        filters={filters}
        setTextFilter={setTextFilter}
        history={history}
    />);

});

test('should render UserManagementPage component correctly with no data no filters', () => {
    expect(wrapper).toMatchSnapshot()
});


test('should render UserManagementPage component correctly with no data and altFilters', () => {
    wrapper.setProps({filters: altFilters});
    expect(wrapper).toMatchSnapshot()
});


test('should handle text change', () => {
    const value = 'name';

    // wrapper.find('Input')

    wrapper.find('Input').simulate('change', {
        target: {value}
    });

    expect(setTextFilter).toHaveBeenLastCalledWith(value);
});

test('modal should be closed by default', () => {
    expect(wrapper.state('modalIsOpen')).toBe(false)
});

test('should handle opening create user window on button click', () => {
    wrapper.find('Button').simulate('click');

    expect(history.push).toHaveBeenLastCalledWith('/create-user');
});