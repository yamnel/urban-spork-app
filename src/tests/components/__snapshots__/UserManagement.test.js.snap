// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render UserManagementPage component correctly with no data and altFilters 1`] = `
<div>
  <div
    className="user-management-nav"
  >
    <Button
      color="success"
      onClick={[Function]}
      tag="button"
    >
      Create User
    </Button>
    <FormGroup
      tag="div"
    >
      <Input
        onChange={[Function]}
        placeholder="Search"
        type="text"
        value="yamnel"
      />
    </FormGroup>
  </div>
  <div>
    <UserTable
      columns={
        Array [
          Object {
            "Header": "Name",
            "accessor": "name",
          },
          Object {
            "Header": "Email",
            "accessor": "email",
          },
          Object {
            "Header": "Department",
            "accessor": "department",
          },
          Object {
            "Header": "Title",
            "accessor": "title",
          },
        ]
      }
      onRowClick={[Function]}
    />
  </div>
  <UserDetailsModal
    isOpen={false}
    toggle={[Function]}
  />
</div>
`;

exports[`should render UserManagementPage component correctly with no data no filters 1`] = `
<div>
  <div
    className="user-management-nav"
  >
    <Button
      color="success"
      onClick={[Function]}
      tag="button"
    >
      Create User
    </Button>
    <FormGroup
      tag="div"
    >
      <Input
        onChange={[Function]}
        placeholder="Search"
        type="text"
        value=""
      />
    </FormGroup>
  </div>
  <div>
    <UserTable
      columns={
        Array [
          Object {
            "Header": "Name",
            "accessor": "name",
          },
          Object {
            "Header": "Email",
            "accessor": "email",
          },
          Object {
            "Header": "Department",
            "accessor": "department",
          },
          Object {
            "Header": "Title",
            "accessor": "title",
          },
        ]
      }
      onRowClick={[Function]}
    />
  </div>
  <UserDetailsModal
    isOpen={false}
    toggle={[Function]}
  />
</div>
`;
